library/src/main/java/com/novoda/downloadmanager/DownloadFile.java

```diff
@@ -60,7 +62,14 @@ void download(Callback callback) {
 
         fileSize.setCurrentSize(filePersistence.getCurrentSize(filePath));
 
-        persistSync();
+        if (downloadFileStatus.isMarkedAsDeleted()) {
+            return;
+        }
+
+        Log.v("persist file " + downloadFileId.rawId() + ", with status: " + downloadFileStatus.status());
+        if (persist()) {
```

Comment:
I don't follow this bit. If it is persisted then return?

Reply:
this is wrong actually, good spot